app/backend/src/couchers/servicers/conversations.py

```diff
@@ -119,6 +118,24 @@ def _get_visible_admins_for_subscription(subscription):
         ]
 
 
+def _user_can_message(session, context, group_chat: GroupChat) -> bool:
+    """
+    User can message in a group chat they are part of if a chat participant exists that:
```

Comment:
I like what you've done here.

On the other hand, I think the logic should probably be uniform for DMs vs Real Group Chats. I.e. if I'm in a group chat with 2 banned users I should still be able to message, because it's an RGC and not a DM.